sympy/solvers/solveset.py

```diff
@@ -978,6 +985,353 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _expo_solver(f):
+    r"""
+    Helper function for solving exponential equations.
+
+    Exponential equations are the type of equations which has a
+    variable to be solved located in the exponent.
+    For example
+
+    .. math:: 5^{(2*x + 3)} - 5^{(3*x - 1)}
+
+    .. math:: 4^{(5 - 9*x)} - 8^{(2 - x)}
+
+    The function evaluates exponential equations having two arguments,
+    i.e, exponential equations having only two exponent terms
+    (like the ones above). Returns `None` for any other type
+    of exponential equation.
+
+    The helper takes the equation to be solved as the only parameter,
+    tries to reduce it to log form (if possible) and returns the
+    modified equation which is further handled by `solveset`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _expo_solver as expo_solver
+    >>> from sympy import symbols
+    >>> x = symbols('x', real=True)
+    >>> expo_solver(3**(2*x) - 2**(x + 3))
+    2*x*log(3) - (x + 3)*log(2)
+
+
+    * Proof of correctness of the method
+
+    The logarithm is the inverse function to exponentiation.
+    The defining relation between exponentiation and logarithm is:
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    Therefore if we are given an equation with exponent terms, we can
+    convert every term to its corresponding log form. This is achieved by
+    taking logarithms and expanding them using log identities so that the
+    equations can be easily handled by `solveset`.
+
+    For example:
+
+    .. math:: 3^{2*x} = 2^{(x + 3)}
+
+    Taking log both sides will reduce the equation to
+
+    .. math:: (2*x)*log(3) = (x + 3)*log(2)
+
+    This form can be easily handed by `solveset`.
+    """
+
+    try:
+        arg1, arg2 = ordered(f.args)
+    except ValueError:
+        return None
+
+    lhs = arg1
+    rhs = -arg2
+
+    lhs = expand_log(log(lhs))
+    rhs = expand_log(log(rhs))
+
+    return lhs - rhs
+
+
+def _check_expo(f, symbol):
+    r"""
+    Helper to check whether an equation is exponential or not.
+
+    It filters out power and `exp` terms from the equation and
```

Comment:
Please explain what do you mean by `exp` and also power. One of them is written as a english word (power) and the other as code (`exp`).